app/src/main/java/com/xa/xpensauditor/MonthWiseVis.java

```diff
@@ -140,30 +143,30 @@ public void onDataChange(@NonNull DataSnapshot datasnapshot) {
                             dumm = snapshot.child("Day").getValue().toString();
                             dumm1 = Integer.valueOf(dumm);
                             System.out.println("dummmmm   " + dumm1 + "    " + dayyy);
-                            if (dayyy == dumm1) {
+                            Date date = new GregorianCalendar(y2, ii-1, dayyy).getTime();
+
+                            if(sortedMap.containsKey(date)){
                                 System.out.println("inside");
-                                int temp = 0;
-                                temp = hm.get(dayyy);
-                                temp = temp + am1;
-                                System.out.println("dummmmm1   " + dayyy + "    " + temp);
-                                hm.put(dayyy, temp);
+                                int temp=0;
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 32, expected level should be one of the following: 24, 26, 28.